code/modules/mob/living/carbon/brain/brain.dm

```diff
@@ -21,13 +21,13 @@
 	. = ..()
 
 /mob/living/carbon/brain/say_understands(mob/speaker, decl/language/speaking)
-	return (issilicon(speaker) && istype(container, /obj/item/mmi)) || ishuman(speaker) || ..()
+	return (issilicon(speaker) && (istype(container, /obj/item/mmi) || istype(loc, /obj/item/organ/internal/posibrain))) || ishuman(speaker) || ..()
```

Comment:
I was going to complain about this not using container, but it turns out that `mmi_holder` sets `container` but `posibrain` doesn't. Works for me.